radiobuttons/src/main/java/com/decathlon/vitamin/compose/radiobuttons/VitaminRadioButtons.kt

```diff
@@ -48,29 +50,41 @@ object VitaminRadioButtons {
         val rowModifier = endContent?.let {
             modifier
                 .clip(RoundedCornerShape(4.dp))
-                .clickable(
-                    enabled = true,
-                    onClick = { onClick?.invoke() },
-                    interactionSource = interactionSource,
-                    indication = LocalIndication.current
-                )
                 .padding(sizes.contentPadding)
         } ?: run { modifier }
+
         val radioColors = RadioButtonDefaults.colors(
             selectedColor = colors.selectedColor,
             unselectedColor = colors.unselectedColor,
             disabledColor = if (selected) colors.disabledSelectedColor
             else colors.disabledUnselectedColor
         )
-        Row(modifier = rowModifier, verticalAlignment = Alignment.CenterVertically) {
-            endContent?.let {
-                RadioButton(
+
+        Row(
+            modifier = rowModifier
+                .selectable(
                     selected = selected,
-                    onClick = null,
                     enabled = enabled,
-                    interactionSource = interactionSource,
-                    colors = radioColors
+                    role = Role.RadioButton,
+                    onClick = {
+                        onClick?.invoke()
+                    }
                 )
+                .semantics(mergeDescendants = true) {},
+            verticalAlignment = Alignment.CenterVertically
+        ) {
+            RadioButton(
+                selected = selected,
+                onClick = {
```

Comment:
Do you really need this callback? The onClick event will be triggered by the `Row`, no?

Reply:
In the implementation of the RadioButton (compose), if the onClick parameter is null, the RadioButton role is not applied. This role allows in particular to define the properties useful for accessibility.

```
// RadioButton.kt (extract)
val selectableModifier =
        if (onClick != null) {
            Modifier.selectable(
                selected = selected,
                onClick = onClick,
                enabled = enabled,
                role = Role.RadioButton,
                interactionSource = interactionSource,
                indication = rememberRipple(
                    bounded = false,
                    radius = RadioButtonRippleRadius
                )
            )
        } else {
            Modifier
        }
```

I will remove the `onClick?.invoke()`